src/index_intf.ml

```diff
@@ -27,28 +27,29 @@ end
 
 module type IO = Io.S
 
-module type MUTEX = sig
-  (** Locks for mutual exclusion *)
+module type SEMAPHORE = sig
+  (** Binary semaphores for mutual exclusion *)
 
   type t
-  (** The type of mutual-exclusion locks. *)
+  (** The type of binary semaphore. *)
 
-  val create : unit -> t
-  (** Return a fresh mutex. *)
+  val make : bool -> t
+  (** Return a fresh semaphore with the given initial state. *)
```

Comment:
can you specify what the initial state means, i.e. `true` for released and `false` for acquired?

Reply:
Good idea; done.